### PR TITLE
Update room summary card header

### PR DESCRIPTION
**Pull request by germain-gg**
For https://github.com/vector-im/element-web/issues/25883

Screenshot 2023-10-31 at 16 31 34
Screenshot 2023-10-31 at 16 31 44
Screenshot 2023-10-31 at 16 34 18

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Update room summary card header ([\#11823](https://github.com/matrix-org/matrix-react-sdk/pull/11823)). Contributed by @germain-gg.

### CODE REVIEW

File: src/components/views/right_panel/RoomSummaryCard.tsx [hunk 1]
```diff
@@ -307,31 +312,74 @@ const RoomSummaryCard: React.FC<IProps> = ({ room, permalinkCreator, onClose, on
...
-
+            <RoomAvatar room={room} size="80px" viewAvatarOnClick />
             <RoomName room={room}>
                 {(name) => (
-                    <h1 className="mx_RoomSummaryCard_roomName" title={name}>
+                    <Heading
+                        as="h1"
+                        size="md"
+                        weight="semibold"
+                        className="mx_RoomSummaryCard_roomName text-primary"
```
Comment (t3chguy) [inline src/components/views/right_panel/RoomSummaryCard.tsx:337]: Specifying these classes feels odd, where are they documented?

File: src/components/views/right_panel/RoomSummaryCard.tsx [hunk 2]
```diff
@@ -307,31 +312,74 @@ const RoomSummaryCard: React.FC<IProps> = ({ room, permalinkCreator, onClose, on
...
-                    </h1>
+                    </Heading>
                 )}
             </RoomName>
-            <div className="mx_RoomSummaryCard_alias" title={alias}>
+            <Text
+                as="div"
+                size="sm"
+                weight="semibold"
+                className="mx_RoomSummaryCard_alias text-secondary"
```
Comment (t3chguy) [inline src/components/views/right_panel/RoomSummaryCard.tsx:348]: Same as above

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/components/views/right_panel/RoomSummaryCard.tsx (hunk 1 above)
Comment (germain-gg) [reply src/components/views/right_panel/RoomSummaryCard.tsx:337]: What documentation would you be interested in having? They're defined as part of `_common.pcss`, i don't think we ever document where CSS classes exist?

Comment (t3chguy) [reply src/components/views/right_panel/RoomSummaryCard.tsx:337]: Sorry, having never seen these classes before I presumed they were from Compound

**Comment by andybalaam** [comment]
All tests that flaked causing this to be removed from the merge queue have been disabled.

**Pull request merged**